agent/engine/containermetadata/write_metadata.go

```
@@ -0,0 +1,267 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/cihub/seelog"
+)
+
+const (
+	inspectContainerTimeout = 30 * time.Second
+	metadataFile            = "metadata.json"
+	mountPoint          = "/ecs/metadata"
+)
+
+// getTaskIDfromArn parses a task Arn and produces the task ID
+func getTaskIDfromArn(taskarn string) string {
```

Why not return an error too ?